numpy/core/src/multiarray/nditer_pywrap.c

```diff
@@ -927,7 +927,7 @@ NpyIter_NestedIters(PyObject *NPY_UNUSED(self),
         }
         for (i = 0; i < nested_naxes[inest]; ++i) {
             PyObject *v = PySequence_GetItem(item, i);
-            npy_intp axis;
+            int axis;
```

Comment:
Actually, this type of mistake crops up a few times - I'll push this to another PR